cou/steps/plan.py

```diff
@@ -354,28 +424,16 @@ async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan
         plan.add_step(
             PreUpgradeStep(
                 description="Backup mysql databases",
-                parallel=False,
                 coro=backup(analysis_result.model),
             )
         )
 
-    control_plane_principal_upgrade_plan = await create_upgrade_group(
-        apps=analysis_result.apps_control_plane,
-        description="Control Plane principal(s) upgrade plan",
-        target=target,
-        force=args.force,
-        filter_function=lambda app: not isinstance(app, SubordinateBaseClass),
-    )
-    plan.add_step(control_plane_principal_upgrade_plan)
-
-    control_plane_subordinate_upgrade_plan = await create_upgrade_group(
-        apps=analysis_result.apps_control_plane,
-        description="Control Plane subordinate(s) upgrade plan",
-        target=target,
-        force=args.force,
-        filter_function=lambda app: isinstance(app, SubordinateBaseClass),
+    plan.sub_steps.extend(
+        _generate_control_plane_plan(target, analysis_result.apps_control_plane, args.force)
     )
-    plan.add_step(control_plane_subordinate_upgrade_plan)
+    # plan.sub_steps.extend(
```

Comment:
I guess you can remove this comment and the `# pragma: no cover`

Reply:
I cannot, since I do not add any tests for this.